app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -140,14 +140,22 @@ def pulp3_enabled?
       end
 
       def pulp3_support?(repository)
-        type = Katello::RepositoryTypeManager.repository_types[repository.content_type]
-        type.pulp3_plugin.present? && pulp3_enabled?
+        pulp3_repository_type_support?(repository.content_type)
       end
 
-      def content_pulp3_support?(content_type)
-        content_type = content_type.is_a?(String) ? content_type : content_type.model_class::CONTENT_TYPE
-        type = Katello::RepositoryTypeManager.find_repository_type content_type
-        type.pulp3_plugin && SmartProxy.pulp_master!.capabilities(PULP3_FEATURE).try(:include?, type.pulp3_plugin)
+      def pulp3_repository_type_support?(repository_type)
+        repository_type_obj = repository_type.is_a?(String) ? Katello::RepositoryTypeManager.repository_types[repository_type] : repository_type
+        fail "Cannot find repository type #{repository_type}, is it enabled?" unless repository_type_obj
+        repository_type_obj.pulp3_plugin.present? && pulp3_enabled? && self.capabilities(PULP3_FEATURE).try(:include?, repository_type_obj.pulp3_plugin)
+      end
+
+      def pulp3_content_support?(content_type)
+        content_type_obj = content_type.is_a?(String) ? RepositoryTypeManager.find_content_type(content_type) : content_type
```

Comment:
`RepositoryTypeManager` here doesn't have `Katello::` like the others do.   Still works but I thought I'd point it out.